src/mount.rs

```diff
@@ -0,0 +1,655 @@
+use lru::LruCache;
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use std::{
+    borrow::Cow,
+    cmp::Ordering,
+    ffi::OsStr,
+    fs,
+    io::{self, ErrorKind, Read},
+    iter::Peekable,
+    num::NonZeroUsize,
+    ops::ControlFlow,
+    path::{Component, Path, PathBuf},
+    sync::{Arc, Mutex},
+    time::Duration,
+};
+use tracing::{error, info, trace};
+
+use bytes::Bytes;
+use itertools::Itertools;
+use tracing::{debug, warn};
+use windows_projfs::{
+    DirectoryEntry, DirectoryInfo, FileInfo, Notification, ProjectedFileSystem,
+    ProjectedFileSystemSource,
+};
+
+use crate::{
+    counters::Counter,
+    monitor::{
+        task::{Task, TaskList},
+        Monitor, Problem,
+    },
+    Apath, Archive, BandId, BandSelectionPolicy, IndexEntry, IndexRead, Kind, Result, StoredTree,
+};
+
+macro_rules! static_dir {
+    ($name:expr) => {
+        DirectoryInfo {
+            directory_name: ($name).to_string(),
+            directory_attributes: DIRECTORY_ATTRIBUTES,
+
+            ..Default::default()
+        }
+        .into()
+    };
+}
+
+struct VoidMonitor;
+impl Monitor for VoidMonitor {
+    fn count(&self, _counter: Counter, _increment: usize) {}
+
+    fn set_counter(&self, _counter: Counter, _value: usize) {}
+
+    fn problem(&self, _problem: Problem) {}
+
+    fn start_task(&self, name: String) -> Task {
+        /*
+         * All data related to the target task will be dropped
+         * as soon the callee drops the task.
+         */
+        let mut list = TaskList::default();
+        list.start_task(name)
+    }
+}
+
+#[derive(Debug)]
+struct HunkMetaInfo {
+    index: u32,
+
+    start_path: Apath,
+    end_path: Apath,
+}
+
+struct HunkHelper {
```

Comment:
This and `StoredFileReader` also don't seem specific to projfs, so could live elsewhere and be tested on Unix.

Reply:
Also, it would help to have comments on the structs.

Reply:
Personally I don't really like the name `HunkHelper` as it's way too generic, but I haven't yet decided on a better one. Shall we may go with `HunkPathIndex`?  
In regards to the comments: you're absolutely right!